File: leetcode_1701/solution.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        initialTime = customers[0][0]
        currentTime = initialTime
        avgWaitTime = 0
        for customer in customers:
            if(currentTime<customer[0]):
                currentTime = customer[0] + customer[1]
            else:
                currentTime = currentTime + customer[1]
            waitTime = currentTime - customer[0]
            avgWaitTime += waitTime
        avgWaitTime = avgWaitTime/(len(customers))
        return avgWaitTime

File: leetcode_1701/solution.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double initialTime{0};
        double currentTime {initialTime};
        double avgWaitTime {0};
        double waitTime {0};
        for (auto& inner_vec : customers)
        {
            if(currentTime<inner_vec[0])
            {
                currentTime = inner_vec[0] + inner_vec[1];
            }
            else
            {
                currentTime += inner_vec[1];
            }
            avgWaitTime += (currentTime - inner_vec[0]);
        }
        avgWaitTime = avgWaitTime/customers.size();
        return avgWaitTime;
    }
};
